projects/plugins/boost/app/modules/optimizations/page-cache/class-cache-preload.php

```diff
@@ -49,6 +51,16 @@ public static function is_available() {
 		return false;
 	}
 
+	/**
+	 * As this is a submodule, this activate is triggered when the parent module is activated,
+	 * despite the module having Is_Always_On.
+	 *
+	 * @since $$next-version$$
+	 */
+	public static function activate() {
+		Setup::get_or_create_instance_of( self::class )->schedule_cornerstone_preload();
```

Comment:
The advantage of having classes that have no side-effects during construction is that you do not always need the same instance. In this case, you can instantiate the class directly and skip adding the `Setup` call.

```suggestion
		$instance = new self();
		$instance->schedule_cornerstone_preload();
```